### ISSUE TITLE
Unable to list resources in a repository using /repositories/:repo_id/resources without specifying a page

### ISSUE THREAD

**Issue by mistydemeo** [archivesspace/archivesspace#129]
I've noticed what seems to be a bug with the `/repositories/:repo_id/resources` endpoint in the API. I'm unable to retrieve a list of resources if I don't specify a page.

This endpoint is [documented](https://archivesspace.github.io/archivesspace/doc/file.API.html#get-repositoriesrepoidresources) as taking no mandatory parameters (beyond the repository); however, when I attempt to use it, it fails with the response reporting missing parameters:

```json
{
  "error": {
    "page": [
      "Parameter required but no value provided"
    ],
    "id_set": [
      "Parameter required but no value provided"
    ],
    "all_ids": [
      "Parameter required but no value provided"
    ]
  }
}
```

If I pass `?page=1`, then the expected response is received. Other endpoints (for example, `/repositories`) don't make `page` mandatory, so I believe this may be a bug. I tested in ArchivesSpace 1.1.2.

**Comment by sdm7g** [comment]
I believe this is intentional. Typically, you might have only a page or two of repositories, but thousands of resources in each repo. 

If you "GET /repositories/:repo_id/resources?all_ids=true”  it will return a list of all of the resource id’s (and only the ids),
which you can then access individually, or as a set, for example:   "http://localhost:8089/repositories/3/resources?id_set=3917,3924,3938” 

You might file it as a bug that the API docs don’t note this explicitly. 


— Steve Majewski

**Comment by cfitz** [comment]
Hi,

Yeah, Steve is right. We recently updated YARD and I just realized some of
the templates for documentation aren't building correctly.

There is a paginated thing here.. you must either supply :page ( with an
optional :page_size param ), :id_set ( with a list of ids to return,
:max_page_size
is set in the AppConfig ), or :all_ids ( which just returns a listing of
all the ids ).

Will look into getting YARD to correctly build the REST endpoint docs.
Sorry about that. b,chris.
